src/components/GoodsList.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+
+type Sort = 'length' | 'alphabet' | 'default';
+
+type Props = {
+  goods: string[],
+};
+
+type State = {
+  isReverse: boolean,
+  sortedBy: Sort,
+  lengthLimit: number,
+};
+
+export class GoodsList extends React.Component<Props, State> {
+  state: Readonly<State> = {
+    isReverse: false,
+    sortedBy: 'default',
+    lengthLimit: 1,
+  };
+
+  handleReverseBtn = () => {
+    this.setState(prev => ({ isReverse: !prev.isReverse }));
+  };
+
+  handleAlphtSortBtn = () => this.sortSwitch('alphabet');
+
+  handleLengthtSortBtn = () => this.sortSwitch('length');
+
+  handleResetBtn = () => {
+    this.setState({ isReverse: false, sortedBy: 'default', lengthLimit: 1 });
+  };
+
+  handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ lengthLimit: +event.target.value });
+  };
+
+  sortSwitch(currentSort: Sort) {
+    this.setState(prev => {
+      if (prev.sortedBy === currentSort) {
+        return { sortedBy: 'default' };
+      }
+
+      return { sortedBy: currentSort };
+    });
+  }
+
+  render(): React.ReactNode {
+    const { isReverse, sortedBy, lengthLimit } = this.state;
+    const preparedList = this.props.goods
+      .filter(good => good.length >= lengthLimit);
+
+    preparedList.sort((g1, g2) => {
+      switch (sortedBy) {
+        case 'alphabet':
+          return g1.localeCompare(g2);
+
+        case 'length':
+          return g1.length - g2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReverse) {
+      preparedList.reverse();
+    }
```

Comment:
This logic shouldn't be here. Method `render` should be as clear as possible.
Let's move this logic to a separate method